File: ruby/isograms.rb
require "set"

test_cases = %w[lumberjacks background downstream six-year-old isograms];

def isogram?(str)
  characters = str.chars.delete_if { |char| char.match(/\-|\ /) }
  characters.size == Set.new(characters).size
end

p test_cases.map { |test_case| isogram?(test_case) }

File: php/isograms.php
<?php

$testCases = ['lumberjacks', 'background', 'downstream', 'six-year-old', 'isograms'];
$results = [];

foreach ($testCases as $testCase)
{
    $chars = str_split(preg_replace('/\ |\-/', '', $testCase));
    $results[] = count($chars) == count(array_unique($chars)) ? 'true' : 'false';
}

echo(implode(', ', $results));

?>
